### ISSUE TITLE
AL09 - Case sensitivity with unquoted identifiers: Clickhouse

### ISSUE THREAD

**Issue by leonardobocci** [sqlfluff/sqlfluff#6019]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

The fix command unexpectedly deletes aliases when ignoring capitalization rules for identifiers.

Submitted query:
SELECT
    myidcol AS col_id,
    my_expected_drop_alias AS my_expected_drop_alias,
    My_Wrong_Case_Col as my_wrong_case_col,-- noqa: CP02
    my_good_col,
FROM
    file('myfile.parquet', 'Parquet')

Reformatted query:
SELECT
    myidcol AS col_id,
    my_expected_drop_alias,
    My_Wrong_Case_Col,-- noqa: CP02
    my_good_col,
FROM
    file('myfile.parquet', 'Parquet')

### Expected Behaviour

Expected result:
SELECT
    myidcol AS col_id,
    my_expected_drop_alias,
    My_Wrong_Case_Col as my_wrong_case_col,-- noqa: CP02
    my_good_col,
FROM
    file('myfile.parquet', 'Parquet')

### Observed Behaviour

When AL09 is used, but CP02 is ignored, AL09 is incorrectly triggered.

### How to reproduce

Example queries provided

### Dialect

Clickhouse

### Version

sqlfluff, version 3.1.0

### Configuration

Default config + inline noqa flag.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by alanmcruickshank** [comment]
@leonardobocci - I'm in the process of resolving this issue (as you can see we've had some conflicting suggestions on how to resolve this). I wonder if you might help me with some details on Clickhouse behaviour? I've had a look at their docs, but it's unclear on how case is treated for identifiers.

If I use your example query:
```sql
SELECT
    myidcol AS col_id,
    my_expected_drop_alias AS my_expected_drop_alias,
    My_Wrong_Case_Col as my_wrong_case_col,-- noqa: CP02
    my_good_col,
FROM
file('myfile.parquet', 'Parquet')
```

You have the line `My_Wrong_Case_Col as my_wrong_case_col`.  Are identifiers (even unquoted ones) always case sensitive? i.e. if this line read just `my_wrong_case_col`, would Clickhouse say the column didn't exist, because only the column reference with the capital letters would work? I'm particularly interested in whether this is only because you're reading from a parquet file or whether this is generally true for tables too.
